overlays/haskell.nix

```diff
@@ -198,11 +198,19 @@ self: super: {
                 HOME=$out cabal new-update cached
             '';
 
+        # Some of features of haskell.nix rely on using a hackage index
+        # to calculate a build plan.  To maintain stabity for caching and
+        # to allow the outputs to be materialized we pin this value here.
+        # If you want to update this value it important to check the
+        # materializations.  Turn `checkMaterialization` on below and
+        # check the CI results before turning it off again.
```

Comment:
:+1:

Something in the dev bit of the user's guide?

Reply:
I have tidied up the messages and updated materialization.md